File: smaller_function_call_test/hello_world.c
// Simple hello world routine as called from fortran

#include <stdio.h>

extern void hello_world();

typedef void (*Caller)(void);

void call_function(Caller c) {
    c();
}





int main() {
    printf("Calling Fortran from C:\n");
    call_function(hello_world);
    return 0;
}

File: smaller_function_call_test/hello_world.f90
! hello world subroutine
module smaller_function_call_test
    use iso_c_binding
    implicit none
contains
    subroutine hello_world() bind(C, name="hello_world")
        print *, "Hello, world!"
    end subroutine hello_world
end module smaller_function_call_test

    